code/game/sound.dm

```diff
@@ -22,6 +26,65 @@
 		0, // Flags (1 = Auto Direct, 2 = Auto Room, 4 = Auto RoomHF)
 	)
 	environment = SOUND_ENVIRONMENT_NONE //Default to none so sounds without overrides dont get reverb
+	var/atom/attached
+	var/list/listeners = list()
+	var/maxdist = 0
+
+/sound/proc/attach( var/atom/towhat )
+	attached = towhat
+	channel = GLOB.channelReserved++
+	if(GLOB.channelReserved>GLOB.channelMax)
+		GLOB.channelReserved=256
+	if(!attached.attachedSounds)
+		attached.attachedSounds = list()
+	attached.attachedSounds.Add(src)
+	GLOB.allAttachedSounds.Add(src)
+
+	recalculate()
+
+/sound/proc/stop()
```

Comment:
All of these procs need to be documented.

https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md